azure-kusto-data/Cargo.toml

```diff
@@ -13,42 +13,42 @@ keywords = ["sdk", "azure", "kusto", "azure-data-explorer"]
 categories = ["api-bindings"]
 
 [dependencies]
-arrow = { version = "31.0.0", optional = true }
-azure_core = { version = "0.8.0", features = [
+arrow = { version = "33.0.0", optional = true }
```

Comment:
Not relevant for this PR, but recently I stumbled across [this](https://github.com/jonhoo/rust-ci-conf/blob/main/.github/workflows/test.yml). Specifically I wanted to look into adopting `-Zminimal-versions`, since keeping arrow-versions up to date in the fast evolving arrow / rust is a bit of a nuisance :)

Reply:
![image](https://user-images.githubusercontent.com/6424271/225253352-c2419869-5d4a-44f7-8ec2-4004af12043d.png)
Not sure how I feel about it, I think while we're in pre-alpha we can break more things